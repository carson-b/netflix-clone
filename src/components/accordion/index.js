/* eslint-disable react-hooks/rules-of-hooks */
/* eslint-disable no-unused-vars */
/* eslint-disable no-shadow */
import React, { createContext, useContext, useState } from 'react';
import {
  Container,
  Frame,
  Inner,
  Item,
  Title,
  Header,
  Body,
} from './styles/accordion';

const ToggleContext = createContext();

export default function accordion({ children, ...restProps }) {
  return (
    <Container {...restProps}>
      <Inner>{children}</Inner>
    </Container>
  );
}

accordion.Title = function accordionTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};

accordion.Frame = function accordionFrame({ children, ...restProps }) {
  return <Frame {...restProps}>{children}</Frame>;
};

accordion.Item = function accordionItem({ children, ...restProps }) {
  const [toggleShow, setToggleShow] = useState(false);
  return (
    <ToggleContext.Provider value={{ toggleShow, setToggleShow }}>
      <Item {...restProps}>{children}</Item>
    </ToggleContext.Provider>
  );
};

accordion.Header = function accordionHeader({ children, ...restProps }) {
  const { toggleShow, setToggleShow } = useContext(ToggleContext);
  return (
    <Header
      onClick={() => setToggleShow((toggleShow) => !toggleShow)}
      {...restProps}
    >
      {children}
      {toggleShow ? (
        <img src="/images/icons/close-slim.png" alt="Close" />
      ) : (
        <img src="/images/icons/add.png" alt="Open" />
      )}
    </Header>
  );
};

accordion.Body = function accordionBody({ children, ...restProps }) {
  const { toggleShow } = useContext(ToggleContext);
  return toggleShow ? <Body {...restProps}>{children}</Body> : null;
};
