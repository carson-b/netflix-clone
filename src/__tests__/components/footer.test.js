import React from 'react';
import { render } from '@testing-library/react';
import { Footer } from '../../components';

describe('<Footer />', () => {
  it('renders the <Footer /> with populated data', () => {
    const { container, getByText } = render(
      <Footer>
        <Footer.Title>Questions? Contact us.</Footer.Title>
        <Footer.Break />
        <Footer.Row>
          <Footer.Column>
            <Footer.Link href="#">FAQ</Footer.Link>
            <Footer.Link>Investor Relations</Footer.Link>
            <Footer.Link>Ways to Watch</Footer.Link>
            <Footer.Link>Corporate Information</Footer.Link>
            <Footer.Link>Netflix Originals</Footer.Link>
          </Footer.Column>

          <Footer.Column>
            <Footer.Link href="#">Help Centre</Footer.Link>
            <Footer.Link>Jobs</Footer.Link>
            <Footer.Link>Terms of Use</Footer.Link>
            <Footer.Link>Contact Us</Footer.Link>
          </Footer.Column>

          <Footer.Column>
            <Footer.Link href="#">Account</Footer.Link>
            <Footer.Link>Redeem Gift Cards</Footer.Link>
            <Footer.Link>Privacy</Footer.Link>
            <Footer.Link>Speed Test</Footer.Link>
          </Footer.Column>

          <Footer.Column>
            <Footer.Link href="#">Media Centre</Footer.Link>
            <Footer.Link>Buy Gift Cards</Footer.Link>
            <Footer.Link>Cookie Preferences</Footer.Link>
            <Footer.Link>Legal Notices</Footer.Link>
          </Footer.Column>
        </Footer.Row>
        <Footer.Break />
        <Footer.Text>Netflix Portugal</Footer.Text>
      </Footer>
    );

    expect(getByText('Questions? Contact us.')).toBeTruthy();
    expect(getByText('FAQ')).toBeTruthy();
    expect(getByText('Investor Relations')).toBeTruthy();
    expect(getByText('Ways to Watch')).toBeTruthy();
    expect(getByText('Corporate Information')).toBeTruthy();
    expect(getByText('Netflix Originals')).toBeTruthy();

    expect(getByText('Help Centre')).toBeTruthy();
    expect(getByText('Jobs')).toBeTruthy();
    expect(getByText('Terms of Use')).toBeTruthy();
    expect(getByText('Contact Us')).toBeTruthy();

    expect(getByText('Account')).toBeTruthy();
    expect(getByText('Redeem Gift Cards')).toBeTruthy();
    expect(getByText('Privacy')).toBeTruthy();
    expect(getByText('Speed Test')).toBeTruthy();
    expect(getByText('Media Centre')).toBeTruthy();
    expect(getByText('Buy Gift Cards')).toBeTruthy();
    expect(getByText('Cookie Preferences')).toBeTruthy();
    expect(getByText('Legal Notices')).toBeTruthy();

    expect(getByText('Netflix Portugal')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
