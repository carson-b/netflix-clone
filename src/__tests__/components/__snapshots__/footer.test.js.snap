// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> renders the <Footer /> with populated data 1`] = `
<div
  class="footer__Container-eh6b4z-0 NSflF"
>
  <p
    class="footer__Title-eh6b4z-4 bQjmyo"
  >
    Questions? Contact us.
  </p>
  <p
    class="footer__Break-eh6b4z-6 beBfaI"
  />
  <div
    class="footer__Row-eh6b4z-2 hRtLtH"
  >
    <div
      class="footer__Column-eh6b4z-1 bVJQyu"
    >
      <a
        class="footer__Link-eh6b4z-3 jVLJmh"
        href="#"
      >
        FAQ
      </a>
      <a
        class="footer__Link-eh6b4z-3 jVLJmh"
      >
        Investor Relations
      </a>
      <a
        class="footer__Link-eh6b4z-3 jVLJmh"
      >
        Ways to Watch
      </a>
      <a
        class="footer__Link-eh6b4z-3 jVLJmh"
      >
        Corporate Information
      </a>
      <a
        class="footer__Link-eh6b4z-3 jVLJmh"
      >
        Netflix Originals
      </a>
    </div>
    <div
      class="footer__Column-eh6b4z-1 bVJQyu"
    >
      <a
        class="footer__Link-eh6b4z-3 jVLJmh"
        href="#"
      >
        Help Centre
      </a>
      <a
        class="footer__Link-eh6b4z-3 jVLJmh"
      >
        Jobs
      </a>
      <a
        class="footer__Link-eh6b4z-3 jVLJmh"
      >
        Terms of Use
      </a>
      <a
        class="footer__Link-eh6b4z-3 jVLJmh"
      >
        Contact Us
      </a>
    </div>
    <div
      class="footer__Column-eh6b4z-1 bVJQyu"
    >
      <a
        class="footer__Link-eh6b4z-3 jVLJmh"
        href="#"
      >
        Account
      </a>
      <a
        class="footer__Link-eh6b4z-3 jVLJmh"
      >
        Redeem Gift Cards
      </a>
      <a
        class="footer__Link-eh6b4z-3 jVLJmh"
      >
        Privacy
      </a>
      <a
        class="footer__Link-eh6b4z-3 jVLJmh"
      >
        Speed Test
      </a>
    </div>
    <div
      class="footer__Column-eh6b4z-1 bVJQyu"
    >
      <a
        class="footer__Link-eh6b4z-3 jVLJmh"
        href="#"
      >
        Media Centre
      </a>
      <a
        class="footer__Link-eh6b4z-3 jVLJmh"
      >
        Buy Gift Cards
      </a>
      <a
        class="footer__Link-eh6b4z-3 jVLJmh"
      >
        Cookie Preferences
      </a>
      <a
        class="footer__Link-eh6b4z-3 jVLJmh"
      >
        Legal Notices
      </a>
    </div>
  </div>
  <p
    class="footer__Break-eh6b4z-6 beBfaI"
  />
  <p
    class="footer__Text-eh6b4z-5 gkOUzd"
  >
    Netflix Portugal
  </p>
</div>
`;
